processor/src/errors.rs

```diff
@@ -620,71 +570,71 @@ pub enum Ext2InttError {
 // ERROR CONTEXT
 // ===============================================================================================
 
-/// Context information to be used when reporting errors.
-#[derive(Debug)]
-pub struct ErrorContext<'a, N: MastNodeExt>(Option<ErrorContextImpl<'a, N>>);
-
-impl<'a, N: MastNodeExt> ErrorContext<'a, N> {
-    /// Creates a new error context for the specified node and source manager.
-    ///
-    /// This method should be used for all nodes except basic block nodes.
-    pub fn new(
-        mast_forest: &'a MastForest,
-        node: &'a N,
-        source_manager: Arc<dyn SourceManager>,
-    ) -> Self {
-        Self(Some(ErrorContextImpl::new(mast_forest, node, source_manager)))
-    }
-
-    /// Creates a new error context for the specified node and source manager.
-    ///
-    /// This method should be used for basic block nodes.
-    pub fn new_with_op_idx(
-        mast_forest: &'a MastForest,
-        node: &'a N,
-        source_manager: Arc<dyn SourceManager>,
-        op_idx: usize,
-    ) -> Self {
-        Self(Some(ErrorContextImpl::new_with_op_idx(
-            mast_forest,
-            node,
-            source_manager,
-            op_idx,
-        )))
-    }
+/// Constructs an error context for the given node in the MAST forest.
+///
+/// When the `debug` feature is enabled, this macro returns an `ErrorContextImpl`; otherwise, it
```

Comment:
Should it be `err_ctx` feature instead? And similarly a few lines below

Reply:
Good catch - fixed